scripts/vale/vale.js

```diff
@@ -146,21 +247,58 @@ function runVale( files ) {
 		} );
 
 		vale.on( 'close', () => {
+			if ( args.save ) {
+				const rawJson = text
+					.replace( valeCommandPattern, '' )
+					.replace( commandInfoLinePattern, '' );
+
+				return resolve( JSON.parse( rawJson ) );
+			}
+
 			const match = text.match( valeFooterPattern );
 
 			if ( !match ) {
 				throw new Error( 'Vale output does not match the footer pattern:\n\n' + text );
 			}
 
-			resolve( {
+			return resolve( {
 				errors: Number( match[ 1 ] ),
 				warnings: Number( match[ 2 ] ),
 				suggestions: Number( match[ 3 ] ),
 				files: Number( match[ 4 ] ),
 				text: text
-					.replace( /^\$ vale .+\n/m, '' )
+					.replace( valeCommandPattern, '' )
 					.replace( valeFooterPattern, '' )
 			} );
 		} );
 	} );
 }
+
+function prepareConfigFiles() {
```

Comment:
I can see that some steps of the entire process are async. Why couldn't it be consistent across the entire script? In other words, if we already use async code, I would substitute all sync calls in favor of async.